dm/tests/dmctl_command/run.sh

```diff
@@ -152,6 +152,261 @@ function run() {
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"validation start --start-time" \
+		"Error: flag needs an argument: --start-time" 1
+
+	echo "--> (fail) validation start: invalid start-time"
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"validation start --start-time xx" \
+		"Error: start-time should be in the format like '2006-01-02 15:04:05' or '2006-01-02T15:04:05'" 1
+
+	echo "--> (fail) validation start: without all-task and task-name"
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"validation start --mode full" \
+		"Error: either \`task-name\` or \`all-task\` should be set" 1
+
+	echo "--> (fail) validation start: with both all-task and task-name"
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"validation start --all-task test" \
+		"Error: either \`task-name\` or \`all-task\` should be set" 1
+
+	echo "--> (fail) validation start: too many arguments"
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"validation start test test2" \
+		"Error: too many arguments are specified" 1
+
+	echo "--> (fail) validation start: non-existed subtask"
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"validation start xxxxx" \
+		"\"result\": false" 1 \
+		"cannot get subtask by task name \`xxxxx\` and sources" 1
+
+	echo "--> (fail) validation start: non-exist source"
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"validation start -s xxxxx --all-task" \
+		"\"result\": false" 1 \
+		"cannot get subtask by sources" 1
+
+	echo "--> (fail) validation stop: without all-task and task-name"
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"validation stop" \
+		"Error: either \`task-name\` or \`all-task\` should be set" 1
+
+	echo "--> (fail) validation stop: with both all-task and task-name"
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"validation stop --all-task test" \
+		"Error: either \`task-name\` or \`all-task\` should be set" 1
+
+	echo "--> (fail) validation stop: too many arguments"
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"validation stop test test2" \
+		"Error: too many arguments are specified" 1
+
+	echo "--> (fail) validation stop: non-existed subtask"
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"validation stop xxxxx" \
+		"\"result\": false" 1 \
+		"cannot get subtask by task name \`xxxxx\` and sources" 1
+
+	echo "--> (fail) validation stop: non-exist source"
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"validation stop -s xxxxx --all-task" \
+		"\"result\": false" 1 \
+		"cannot get subtask by sources" 1
+
+	echo "--> (fail) validation stop: stop not-enabled validator"
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"validation stop -s $SOURCE_ID1 test" \
+		"\"result\": false" 1 \
+		"some target validator(test with source $SOURCE_ID1) is not enabled" 1
+
+	echo "--> (success) validation start: start validation without explicit mode for source 1"
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"validation start -s $SOURCE_ID1 test" \
```

Comment:
Could you add a test case that specifies multiple sources? More than one source is valid too.

Reply:
https://github.com/pingcap/tiflow/pull/5534/commits/fed53e214610d2cc6469aa613e4db9e675c9366c